docs/reference/protocol-spec.md

```diff
@@ -218,28 +255,34 @@ integers and a bulk string can be encoded as follows:
     $5\r\n
     hello\r\n
 
-(The reply was split into multiple lines for clarity).
+(The raw RESP encoding is split into multiple lines for readability).
+
+The first line the server sent is `*5\r\n`.
+This numeric value tells the client that five reply types are about to follow it.
+Then, every successive reply constitutes an element in the array.
 
-The first line the server sent is `*5\r\n` in order to specify that five
-replies will follow. Then every reply constituting the items of the
-Multi Bulk reply are transmitted.
+{{% alert title="Multi Bulk Reply" color="info" %}}
+In some places, the RESP Array type may be referred to as **Multi Bulk**.
+The two are the same.
+{{% /alert %}}
 
-Null Arrays exist as well and are an alternative way to
-specify a Null value (usually the Null Bulk String is used, but for historical
-reasons we have two formats).
+<a name="nil-array-reply"></a>
 
-For instance, when the `BLPOP` command times out, it returns a Null Array
-that has a count of `-1` as in the following example:
+#### Null Arrays
+Whereas RESP3 has a dedicated data type for [null values](#nulls), RESP2 has no such type. Instead, due to historical reasons, the representation of null values in RESP2 is via predetermined forms of the [Bulk Strings](#bulk-strings) and Arrays types.
 
-    "*-1\r\n"
+Null Arrays exist as an alternative way of representing a Null value.
+For instance, when the `BLPOP` command times out, it returns a Null Array.
 
-A client library API should return a null object and not an empty Array when
-Redis replies with a Null Array. This is necessary to distinguish
-between an empty list and a different condition (for instance the timeout
-condition of the `BLPOP` command).
+The encoding of a Null Array is that of an Array with the length of -1, i.e.:
 
-Nested arrays are possible in RESP. For example a nested array of two arrays
-is encoded as follows:
+    *-1\r\n
+
+A client should return its caller a null object when Redis replies with a Null Array, rather than an empty array.
+This is necessary to distinguish between an empty list and a different condition (for instance, the timeout condition of the `BLPOP` command).
+
+Nested arrays are possible in RESP.
```

Comment:
Arrays can contain any nested data type, not just arrays

Reply:
All aggregates can do that, but I understand your point and have added some words to that effect.